x-pack/solutions/observability/plugins/observability/public/pages/alert_details/components/related_dashboards/dashboard_tile.tsx

```diff
@@ -35,20 +32,28 @@ export function DashboardTile({
   dashboard,
   actionButtonProps,
 }: {
-  dashboard: DashboardMetadata;
+  dashboard: RelatedDashboard;
   actionButtonProps?: ActionButtonProps;
 }) {
   const {
     services: {
       share: { url: urlService },
+      savedObjectsTagging: { ui: savedObjectsTaggingUi },
     },
   } = useKibana();
   const dashboardLocator = urlService.locators.get<DashboardLocatorParams>(DASHBOARD_APP_LOCATOR);
 
+  const tagsReferences: SavedObjectsReference[] = dashboard.tags?.length
+    ? dashboard.tags.reduce((acc, value) => {
```

Comment:
Nothing really against reduce here. This looks a bit more readable to me and avoids repeated array spreads.

```
(dashboard.tags || []).flatMap(tag => {
  const ref = savedObjectsTaggingUi.convertNameToReference(tag);
  return ref ? [{ ...ref, name: tag }] : [];
});
```

Reply:
Changed, I like this implementation more.